modular_skyrat/modules/nuclear_reactor/code/reactor.dm

```diff
@@ -0,0 +1,410 @@
+	var/pressure_modifier = core_pressure
+
+	if(core_temperature <= REACTOR_TEMPERATURE_MINIMUM) // We are going to drop  pressure as steam is now turning into water.
+		pressure_modifier -= REACTOR_PRESSURE_CONDENSATION_RATE * core_temperature
+	else // We are going to increase pressure as water is now turning into steam.
+		pressure_modifier += REACTOR_PRESSURE_EVAPORATION_RATE * core_temperature
+
+	core_pressure = clamp(pressure_modifier, 0, INFINITY) // We cannot go below zero pressure.
+
+/obj/machinery/reactor/proc/get_cooling_potential()
+	if(!reactor_heat_exchanger)
+		var/turf/our_turf = get_turf(src)
+		return our_turf.temperature //If we do not have a heat exchanger, we just return the temperature of our turf.
+
+	calculated_cooling_potential = reactor_heat_exchanger.calculated_cooling_potential
+
+/obj/machinery/reactor/proc/move_control_rods()
+	if(!LAZYLEN(control_rods)) //We have no control rods to move... don't bother trying.
+		return
+	for(var/obj/item/reactor_control_rod/control_rod in control_rods)
+		var/control_rod_position_delta = desired_control_rod_insertion - control_rod.insertion_percent
+		var/control_rod_movement_amount = clamp(control_rod_position_delta, -control_rod.movement_rate, control_rod.movement_rate)
+		control_rod.insertion_percent += clamp(control_rod_movement_amount, 0, 100)
+
+/obj/machinery/reactor/proc/calculate_power()
+	if(!LAZYLEN(fuel_rods)) //We have no fuel rods to calculate power from.
+		return 0
+	var/power = 0
+	// Overall reactor power is the sum of the power of all the fuel rods.
+	for(var/obj/item/reactor_fuel_rod/fuel_rod in fuel_rods)
+		power += fuel_rod.radioactivity
+	if(reactor_moderator)
+		power *= reactor_moderator.moderator_efficency
+	if(calculated_control_rod_efficiency)
+		power /= (calculated_control_rod_efficiency * 0.5)
+	return power
+
+/obj/machinery/reactor/proc/calculate_control_rod_efficiency()
+	var/efficency = 0
+	for(var/obj/item/reactor_control_rod/control_rod in control_rods)
+		efficency += (control_rod.efficency * control_rod.insertion_percent)
+	return efficency
+
+/obj/machinery/reactor/proc/process_temperature()
+	// The difference between our lowest possible temperature(target) and our core temperature. Positive if heating, negative if cooling.
+	var/calculated_temperature_delta = calculated_cooling_potential - core_temperature
+
+	var/calculated_temperature_modifier = clamp(calculated_temperature_delta, -REACTOR_MAX_TEMPERATURE_CHANGE, REACTOR_MAX_TEMPERATURE_CHANGE)
+
+	// We calculate the heat addition from the power of the reactor.
+	// If you can get the cooling potential to be equal to the power, then the reactor temp is stable.
+	calculated_temperature_modifier += calculated_power
+
+	// Now we add our ambient temperature to the mix.
+	var/turf/our_turf = get_turf(src)
+
+	var/calculated_ambient_temperature_delta = our_turf.temperature - core_temperature
```

Comment:
I'm pretty sure temperature needs to be found from the gas/air mix or environment of the turf rather than the turf directly? 

You can get the air mix by doing return_air() on the turf:
> var/datum/gas_mixture/enviroment = turf.return_air()

then do environment.temperature etc.

Reply:
you're right